src/corefx/System.Globalization.Native/localeNumberData.cpp

```diff
@@ -367,14 +379,73 @@ extern "C" int32_t GetLocaleInfoInt(const UChar* localeName, LocaleNumberData lo
 		*value = GetCurrencyNegativePattern(locale);
 		break;
 	case CalendarType:
-		*value = 1; //TODO: implement
+	{
+		LocalPointer<Calendar> calendar(Calendar::createInstance(locale, status));
```

Comment:
I think the logic for this should be different.  Instead of failing if the default calendar isn't supported, we should get the list of supported calendars and start enumerating them, returning the first supported calendar in the list.
To get the list of supported calendars, see the GetCalendars method in calendarData.cpp.